### ISSUE TITLE
[0.9.4.1] "panic: runtime error" after dropping a retention policy used by continuous query 

### ISSUE THREAD

**Issue by jbothma** [influxdb/influxdb#4230]
If I drop a retention policy that's in use by a continuous query, influx dies with

```panic: runtime error: invalid memory address or nil pointer dereference```

**Comment by jbothma** [comment]
I lost the full stack trace now, sorry

**Comment by beckettsean** [comment]
@jbothma what version of InfluxDB?

**Comment by jbothma** [comment]
0.9.4.1

**Comment by madshall** [comment]
Same for me. 0.9.4.2 The full log is here:
```bash
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x20 pc=0x62a5aa]

goroutine 58 [running]:
github.com/influxdb/influxdb/cluster.(*PointsWriter).MapShards(0xc08204f000, 0xc0880845c0, 0x9, 0x0, 0x0)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/cluster/points_writer.go:179 +0x1fa
github.com/influxdb/influxdb/cluster.(*PointsWriter).WritePoints(0xc08204f000, 0xc0880845c0, 0x0, 0x0)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/cluster/points_writer.go:215 +0x1e0
github.com/influxdb/influxdb/services/continuous_querier.(*Service).runContinuousQueryAndWriteResult(0xc08205efc0, 0xc0878be480, 0x0, 0x0)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/services/continuous_querier/service.go:385 +0x945
github.com/influxdb/influxdb/services/continuous_querier.(*Service).ExecuteContinuousQuery(0xc08205efc0, 0xc086bdfce8, 0xc0860e2000, 0xecdb1e68d, 0x880d3f8, 0x1008060, 0x0, 0x0)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/services/continuous_querier/service.go:292 +0x8a9
github.com/influxdb/influxdb/services/continuous_querier.(*Service).runContinuousQueries(0xc08205efc0, 0xc086bdfe88)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/services/continuous_querier/service.go:224 +0x3c7
github.com/influxdb/influxdb/services/continuous_querier.(*Service).backgroundLoop(0xc08205efc0)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/services/continuous_querier/service.go:203 +0x54a
created by github.com/influxdb/influxdb/services/continuous_querier.(*Service).Open
	/Users/corylanou/go/src/github.com/influxdb/influxdb/services/continuous_querier/service.go:126 +0x2ab

goroutine 1 [chan receive]:
main.(*Main).Run(0xc085dd1f10, 0xc082009fc0, 0x2, 0x2, 0x0, 0x0)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/cmd/influxd/main.go:91 +0xa1d
main.main()
	/Users/corylanou/go/src/github.com/influxdb/influxdb/cmd/influxd/main.go:42 +0x2a2

goroutine 5 [syscall]:
os/signal.loop()
	/usr/local/Cellar/go/1.5.1/libexec/src/os/signal/signal_unix.go:22 +0x1f
created by os/signal.init.1
	/usr/local/Cellar/go/1.5.1/libexec/src/os/signal/signal_unix.go:28 +0x3e

goroutine 8 [IO wait]:
net.runtime_pollWait(0x3969e0, 0x72, 0xc08211e010)
	/usr/local/Cellar/go/1.5.1/libexec/src/runtime/netpoll.go:157 +0x67
net.(*pollDesc).Wait(0xc08201b070, 0x72, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/fd_poll_runtime.go:73 +0x41
net.(*ioSrv).ExecIO(0xc082024128, 0xc08201af60, 0xbe6e10, 0x8, 0xc08211c000, 0xc08211a000, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/fd_windows.go:182 +0x177
net.(*netFD).acceptOne(0xc08201af00, 0xc082118000, 0x2, 0x2, 0xc08201af60, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/fd_windows.go:564 +0x26c
net.(*netFD).accept(0xc08201af00, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/fd_windows.go:594 +0x173
net.(*TCPListener).AcceptTCP(0xc082024138, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/tcpsock_posix.go:254 +0x54
net.(*TCPListener).Accept(0xc082024138, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/tcpsock_posix.go:264 +0x44
github.com/influxdb/influxdb/tcp.(*Mux).Serve(0xc0820fb700, 0x395a60, 0xc082024138, 0x0, 0x0)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tcp/mux.go:48 +0x52
created by github.com/influxdb/influxdb/cmd/influxd/run.(*Server).Open.func1
	/Users/corylanou/go/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:346 +0xaf0

goroutine 9 [chan receive]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc082024140, 0x0, 0x0, 0x0, 0x0)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tcp/mux.go:129 +0x76
github.com/influxdb/influxdb/meta.(*raftLayer).Accept(0xc082103710, 0x0, 0x0, 0x0, 0x0)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/meta/store.go:2059 +0x58
github.com/hashicorp/raft.(*NetworkTransport).listen(0xc082031500)
	/Users/corylanou/go/src/github.com/hashicorp/raft/net_transport.go:346 +0x80
created by github.com/hashicorp/raft.NewNetworkTransport
	/Users/corylanou/go/src/github.com/hashicorp/raft/net_transport.go:138 +0x28b

goroutine 10 [select]:
github.com/hashicorp/raft.(*Raft).leaderLoop(0xc08207e1c0)
	/Users/corylanou/go/src/github.com/hashicorp/raft/raft.go:849 +0xcd5
github.com/hashicorp/raft.(*Raft).runLeader(0xc08207e1c0)
	/Users/corylanou/go/src/github.com/hashicorp/raft/raft.go:821 +0x93d
github.com/hashicorp/raft.(*Raft).run(0xc08207e1c0)
	/Users/corylanou/go/src/github.com/hashicorp/raft/raft.go:602 +0xc2
github.com/hashicorp/raft.(*Raft).(github.com/hashicorp/raft.run)-fm()
	/Users/corylanou/go/src/github.com/hashicorp/raft/raft.go:252 +0x27
github.com/hashicorp/raft.(*raftState).goFunc.func1(0xc08207e1c0, 0xc082114770)
	/Users/corylanou/go/src/github.com/hashicorp/raft/state.go:152 +0x4b
created by github.com/hashicorp/raft.(*raftState).goFunc
	/Users/corylanou/go/src/github.com/hashicorp/raft/state.go:153 +0x54

goroutine 11 [select]:
github.com/hashicorp/raft.(*Raft).runFSM(0xc08207e1c0)
	/Users/corylanou/go/src/github.com/hashicorp/raft/raft.go:509 +0x10f2
github.com/hashicorp/raft.(*Raft).(github.com/hashicorp/raft.runFSM)-fm()
	/Users/corylanou/go/src/github.com/hashicorp/raft/raft.go:253 +0x27
github.com/hashicorp/raft.(*raftState).goFunc.func1(0xc08207e1c0, 0xc082114780)
	/Users/corylanou/go/src/github.com/hashicorp/raft/state.go:152 +0x4b
created by github.com/hashicorp/raft.(*raftState).goFunc
	/Users/corylanou/go/src/github.com/hashicorp/raft/state.go:153 +0x54

goroutine 12 [select]:
github.com/hashicorp/raft.(*Raft).runSnapshots(0xc08207e1c0)
	/Users/corylanou/go/src/github.com/hashicorp/raft/raft.go:1636 +0x49a
github.com/hashicorp/raft.(*Raft).(github.com/hashicorp/raft.runSnapshots)-fm()
	/Users/corylanou/go/src/github.com/hashicorp/raft/raft.go:254 +0x27
github.com/hashicorp/raft.(*raftState).goFunc.func1(0xc08207e1c0, 0xc082114790)
	/Users/corylanou/go/src/github.com/hashicorp/raft/state.go:152 +0x4b
created by github.com/hashicorp/raft.(*raftState).goFunc
	/Users/corylanou/go/src/github.com/hashicorp/raft/state.go:153 +0x54

goroutine 13 [select]:
github.com/influxdb/influxdb/meta.(*localRaft).logLeaderChanges(0xc082007f40)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/meta/state.go:187 +0x4fc
created by github.com/influxdb/influxdb/meta.(*localRaft).open
	/Users/corylanou/go/src/github.com/influxdb/influxdb/meta/state.go:177 +0x134b

goroutine 14 [chan receive]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc082024148, 0x0, 0x0, 0x0, 0x0)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tcp/mux.go:129 +0x76
github.com/influxdb/influxdb/meta.(*Store).serveExecListener(0xc0820708c0)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/meta/store.go:622 +0xa9
created by github.com/influxdb/influxdb/meta.(*Store).Open
	/Users/corylanou/go/src/github.com/influxdb/influxdb/meta/store.go:239 +0x36b

goroutine 15 [chan receive]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc082024150, 0x0, 0x0, 0x0, 0x0)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tcp/mux.go:129 +0x76
github.com/influxdb/influxdb/meta.(*Store).serveRPCListener(0xc0820708c0)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/meta/store.go:726 +0xa9
created by github.com/influxdb/influxdb/meta.(*Store).Open
	/Users/corylanou/go/src/github.com/influxdb/influxdb/meta/store.go:242 +0x3bd

goroutine 35 [select]:
github.com/influxdb/influxdb/cmd/influxd/run.(*Server).monitorErrorChan(0xc0820eb200, 0xc08200cd20)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:492 +0x14c
created by github.com/influxdb/influxdb/cmd/influxd/run.(*Server).Open.func1
	/Users/corylanou/go/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:352 +0xc6e

goroutine 36 [select]:
github.com/influxdb/influxdb/monitor.(*Monitor).storeStatistics(0xc08210a1a0)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/monitor/service.go:333 +0x1883
created by github.com/influxdb/influxdb/monitor.(*Monitor).Open
	/Users/corylanou/go/src/github.com/influxdb/influxdb/monitor/service.go:134 +0x313

goroutine 20 [select]:
github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).autoflusher(0xc0821180e0, 0xc0821763c0)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:604 +0x56e
created by github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).Open
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:240 +0x62e

goroutine 37 [select]:
github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).autoflusher(0xc0821181c0, 0xc0821466c0)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:604 +0x56e
created by github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).Open
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:240 +0x62e

goroutine 38 [select]:
github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).autoflusher(0xc0821182a0, 0xc082064240)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:604 +0x56e
created by github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).Open
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:240 +0x62e

goroutine 21 [select]:
github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).autoflusher(0xc082118460, 0xc082126840)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:604 +0x56e
created by github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).Open
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:240 +0x62e

goroutine 22 [select]:
github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).autoflusher(0xc082289dc0, 0xc0821268a0)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:604 +0x56e
created by github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).Open
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:240 +0x62e

goroutine 39 [select]:
github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).autoflusher(0xc082182000, 0xc082176900)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:604 +0x56e
created by github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).Open
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:240 +0x62e

goroutine 23 [select]:
github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).autoflusher(0xc0821821c0, 0xc082176780)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:604 +0x56e
created by github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).Open
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:240 +0x62e

goroutine 40 [select]:
github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).autoflusher(0xc082182380, 0xc082064060)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:604 +0x56e
created by github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).Open
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:240 +0x62e

goroutine 41 [select]:
github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).autoflusher(0xc082182540, 0xc082064ba0)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:604 +0x56e
created by github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).Open
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:240 +0x62e

goroutine 24 [select]:
github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).autoflusher(0xc082182700, 0xc082127260)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:604 +0x56e
created by github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).Open
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:240 +0x62e

goroutine 25 [select]:
github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).autoflusher(0xc0828efdc0, 0xc0829ad4a0)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:604 +0x56e
created by github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).Open
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:240 +0x62e

goroutine 50 [select]:
github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).autoflusher(0xc082cdd960, 0xc086eae480)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:604 +0x56e
created by github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).Open
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:240 +0x62e

goroutine 51 [select]:
github.com/influxdb/influxdb/services/hh.(*Service).retryWrites(0xc08205ef30)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/services/hh/service.go:103 +0x31b
created by github.com/influxdb/influxdb/services/hh.(*Service).Open
	/Users/corylanou/go/src/github.com/influxdb/influxdb/services/hh/service.go:67 +0x217

goroutine 52 [select]:
github.com/influxdb/influxdb/services/hh.(*Service).expireWrites(0xc08205ef30)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/services/hh/service.go:121 +0x2d8
created by github.com/influxdb/influxdb/services/hh.(*Service).Open
	/Users/corylanou/go/src/github.com/influxdb/influxdb/services/hh/service.go:68 +0x239

goroutine 53 [chan receive]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc082024158, 0x0, 0x0, 0x0, 0x0)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tcp/mux.go:129 +0x76
github.com/influxdb/influxdb/cluster.(*Service).serve(0xc082031420)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/cluster/service.go:84 +0xe2
created by github.com/influxdb/influxdb/cluster.(*Service).Open
	/Users/corylanou/go/src/github.com/influxdb/influxdb/cluster/service.go:61 +0x13c

goroutine 54 [select]:
github.com/influxdb/influxdb/services/precreator.(*Service).runPrecreation(0xc0820fb3c0)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/services/precreator/service.go:75 +0x456
created by github.com/influxdb/influxdb/services/precreator.(*Service).Open
	/Users/corylanou/go/src/github.com/influxdb/influxdb/services/precreator/service.go:53 +0x226

goroutine 55 [chan receive]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc082024160, 0x0, 0x0, 0x0, 0x0)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tcp/mux.go:129 +0x76
github.com/influxdb/influxdb/services/snapshotter.(*Service).serve(0xc082007bd0)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/services/snapshotter/service.go:75 +0xb0
created by github.com/influxdb/influxdb/services/snapshotter.(*Service).Open
	/Users/corylanou/go/src/github.com/influxdb/influxdb/services/snapshotter/service.go:48 +0x137

goroutine 56 [chan receive]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc082024168, 0x0, 0x0, 0x0, 0x0)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/tcp/mux.go:129 +0x76
github.com/influxdb/influxdb/services/copier.(*Service).serve(0xc0820fb440)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/services/copier/service.go:78 +0xb0
created by github.com/influxdb/influxdb/services/copier.(*Service).Open
	/Users/corylanou/go/src/github.com/influxdb/influxdb/services/copier/service.go:51 +0x137

goroutine 57 [IO wait]:
net.runtime_pollWait(0x396860, 0x72, 0xc084bf1490)
	/usr/local/Cellar/go/1.5.1/libexec/src/runtime/netpoll.go:157 +0x67
net.(*pollDesc).Wait(0xc0879ec2f0, 0x72, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/fd_poll_runtime.go:73 +0x41
net.(*ioSrv).ExecIO(0xc082024128, 0xc0879ec1e0, 0xbe6e10, 0x8, 0xc083060040, 0xc0879ecd80, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/fd_windows.go:182 +0x177
net.(*netFD).acceptOne(0xc0879ec180, 0xc08342a000, 0x2, 0x2, 0xc0879ec1e0, 0xc083060021, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/fd_windows.go:564 +0x26c
net.(*netFD).accept(0xc0879ec180, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/fd_windows.go:594 +0x173
net.(*TCPListener).AcceptTCP(0xc082024058, 0x455a80, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/tcpsock_posix.go:254 +0x54
net.(*TCPListener).Accept(0xc082024058, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/tcpsock_posix.go:264 +0x44
net/http.(*Server).Serve(0xc086eae8a0, 0x395a60, 0xc082024058, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/http/server.go:1887 +0xba
net/http.Serve(0x395a60, 0xc082024058, 0x34cfe30, 0xc084bf1480, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/http/server.go:1765 +0x8c
github.com/influxdb/influxdb/services/admin.(*Service).serve(0xc082007cc0)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/services/admin/service.go:107 +0x136
created by github.com/influxdb/influxdb/services/admin.(*Service).Open
	/Users/corylanou/go/src/github.com/influxdb/influxdb/services/admin/service.go:70 +0x498

goroutine 59 [IO wait]:
net.runtime_pollWait(0x3967a0, 0x72, 0xc0839dc6e0)
	/usr/local/Cellar/go/1.5.1/libexec/src/runtime/netpoll.go:157 +0x67
net.(*pollDesc).Wait(0xc0879ec770, 0x72, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/fd_poll_runtime.go:73 +0x41
net.(*ioSrv).ExecIO(0xc082024128, 0xc0879ec660, 0xbe6e10, 0x8, 0xc082e48660, 0xc086f90300, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/fd_windows.go:182 +0x177
net.(*netFD).acceptOne(0xc0879ec600, 0xc086b2c0e0, 0x2, 0x2, 0xc0879ec660, 0x220000, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/fd_windows.go:564 +0x26c
net.(*netFD).accept(0xc0879ec600, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/fd_windows.go:594 +0x173
net.(*TCPListener).AcceptTCP(0xc082024060, 0xc087067db0, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/tcpsock_posix.go:254 +0x54
net.(*TCPListener).Accept(0xc082024060, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/tcpsock_posix.go:264 +0x44
net/http.(*Server).Serve(0xc086eae960, 0x395a60, 0xc082024060, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/http/server.go:1887 +0xba
net/http.Serve(0x395a60, 0xc082024060, 0x34cfe58, 0xc08204f180, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/http/server.go:1765 +0x8c
github.com/influxdb/influxdb/services/httpd.(*Service).serve(0xc08200d320)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/services/httpd/service.go:133 +0x75
created by github.com/influxdb/influxdb/services/httpd.(*Service).Open
	/Users/corylanou/go/src/github.com/influxdb/influxdb/services/httpd/service.go:101 +0x6d1

goroutine 60 [select]:
github.com/influxdb/influxdb/services/retention.(*Service).deleteShardGroups(0xc082007e50)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/services/retention/service.go:69 +0x358
created by github.com/influxdb/influxdb/services/retention.(*Service).Open
	/Users/corylanou/go/src/github.com/influxdb/influxdb/services/retention/service.go:45 +0x1bb

goroutine 61 [select]:
github.com/influxdb/influxdb/services/retention.(*Service).deleteShards(0xc082007e50)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/services/retention/service.go:102 +0x7a7
created by github.com/influxdb/influxdb/services/retention.(*Service).Open
	/Users/corylanou/go/src/github.com/influxdb/influxdb/services/retention/service.go:46 +0x1e0

goroutine 62 [select]:
github.com/influxdb/influxdb/cmd/influxd/run.(*Command).monitorServerErrors(0xc08204e400)
	/Users/corylanou/go/src/github.com/influxdb/influxdb/cmd/influxd/run/command.go:138 +0x2cc
created by github.com/influxdb/influxdb/cmd/influxd/run.(*Command).Run
	/Users/corylanou/go/src/github.com/influxdb/influxdb/cmd/influxd/run/command.go:120 +0xef2

goroutine 26 [IO wait]:
net.runtime_pollWait(0x396620, 0x72, 0xc084db4140)
	/usr/local/Cellar/go/1.5.1/libexec/src/runtime/netpoll.go:157 +0x67
net.(*pollDesc).Wait(0xc0879ed070, 0x72, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/fd_poll_runtime.go:73 +0x41
net.(*ioSrv).ExecIO(0xc082024128, 0xc0879ecf60, 0xbdcc48, 0x7, 0xd26228, 0xc0821161c0, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/fd_windows.go:182 +0x177
net.(*netFD).Read(0xc0879ecf00, 0xc085de1000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/fd_windows.go:463 +0x17e
net.(*conn).Read(0xc0824b61e0, 0xc085de1000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/net.go:172 +0xeb
net/http.(*liveSwitchReader).Read(0xc083582048, 0xc085de1000, 0x1000, 0x1000, 0x2c0, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/http/server.go:219 +0xab
io.(*LimitedReader).Read(0xc082e48620, 0xc085de1000, 0x1000, 0x1000, 0xc082027588, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/io/io.go:427 +0xc4
bufio.(*Reader).fill(0xc086eaf6e0)
	/usr/local/Cellar/go/1.5.1/libexec/src/bufio/bufio.go:97 +0x1f0
bufio.(*Reader).ReadSlice(0xc086eaf6e0, 0xc081da690a, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/bufio/bufio.go:328 +0x221
bufio.(*Reader).ReadLine(0xc086eaf6e0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/bufio/bufio.go:357 +0x5a
net/textproto.(*Reader).readLineSlice(0xc0848009f0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/textproto/reader.go:55 +0x88
net/textproto.(*Reader).ReadLine(0xc0848009f0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/textproto/reader.go:36 +0x47
net/http.ReadRequest(0xc086eaf6e0, 0xc086b2c7e0, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/http/request.go:653 +0xbd
net/http.(*conn).readRequest(0xc083582000, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/http/server.go:633 +0x336
net/http.(*conn).serve(0xc083582000)
	/usr/local/Cellar/go/1.5.1/libexec/src/net/http/server.go:1319 +0x72e
created by net/http.(*Server).Serve
	/usr/local/Cellar/go/1.5.1/libexec/src/net/http/server.go:1910 +0x3fd

```

**Comment by otoolep** [comment]
Fixed in the upcoming 0.9.5 release by https://github.com/influxdb/influxdb/pull/4136
